integration-tests/tests/run-notebook.js

```diff
@@ -26,6 +26,9 @@ const testRunNotebookFn = _.flow(
   await click(page, clickable({ text: 'Create Notebook' }))
   await click(page, clickable({ textContains: notebookName }))
   await waitForNoSpinners(page)
+  //Adding a short microsleep to make sure that we finish waiting for the spinner
+  //Make sure that the edit button is clicked
+  await delay(500)
```

Comment:
My hypothesis is that the problem is caused by this sequence of events:

1. Puppeteer test clicks notebook
2. Terra renders Edit button
3. Puppeteer test sees Edit button
4. Terra renders spinner
5. Puppeteer test clicks where the Edit button is located, but the click is swallowed by the spinner
6. Terra removes spinner and renders notebook preview
7. Test times out and screenshot shows notebook preview

If that's the case, then the test needs to micro-sleep _before_ `waitForNoSpinners`.

I know that I'm being very particular about this, but I think we need to be very precise in both the execution and documentation of these fixes. We have to leave enough context so that the next person who has to change the test (because something about the page behavior changed) understands whether or not the reason for a micro-sleep is still valid. If they have to guess and guess wrong, they will either re-introduce the flakiness or leave unnecessary sleeps in the tests, both of which lead to test suite pain down the road. (Believe me, this is the 4th time I've witnessed/participated in the evolution of a suite of browser-driven tests.)